packages/core/src/messages/responses/item-type-api.ts

```diff
@@ -0,0 +1,76 @@
+import {SdkError} from '../../errors/errors';
+import {
+  ResponseBase,
+  ResponseError,
+  ResponseHit,
+  ResponseMiss,
+} from './response-base';
+
+/**
+ * Parent response type for a item type request.  The
+ * response object is resolved to a type-safe object of one of
+ * the following subtypes:
+ *
+ * - {Hit}
+ * - {Miss}
+ * - {Error}
+ *
+ * `instanceof` type guards can be used to operate on the appropriate subtype.
+ * @example
+ * For example:
+ * ```
+ * if (response instanceof ItemType.Error) {
+ *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
+ *   // `ItemType.Error` in this block, so you will have access to the properties
+ *   // of the Error class; e.g. `response.errorCode()`.
+ * }
+ * ```
+ */
+export abstract class Response extends ResponseBase {}
+
+class _Hit extends Response {
+  private readonly keyType: string;
+
+  constructor(keyType: string) {
```

Comment:
I'm just wondering, should this `keyType` be an enum? Probably not because it would require us to update this enum if we returned new data types? And I could see us forgetting to do that. But it would make working with this api in typescript easier